projects/kubernetes-sigs/cluster-api/patches/0002-Add-unstacked-etcd-support.patch

```diff
@@ -629,30 +628,38 @@ index 131347515..098de41e6 100644
 +		}
 +	}
 +
- 	// Add finalizer first if not exist to avoid the race condition between init and delete
- 	if !controllerutil.ContainsFinalizer(kcp, controlplanev1.KubeadmControlPlaneFinalizer) {
- 		controllerutil.AddFinalizer(kcp, controlplanev1.KubeadmControlPlaneFinalizer)
-@@ -475,6 +506,15 @@ func (r *KubeadmControlPlaneReconciler) reconcileDelete(ctx context.Context, clu
- 	}
- 	ownedMachines := allMachines.Filter(collections.OwnedMachines(kcp))
+ 	// Add finalizer first if not set to avoid the race condition between init and delete.
+ 	// Note: Finalizers in general can only be added when the deletionTimestamp is not set.
+ 	if kcp.ObjectMeta.DeletionTimestamp.IsZero() && !controllerutil.ContainsFinalizer(kcp, controlplanev1.KubeadmControlPlaneFinalizer) {
```

Comment:
Why do we need to change this code? It doesn't seem like it was ever part of out patches.

Reply:
Ahhh is this just part of the original code? And the patch only changes to show that code around our actual code additions?

Reply:
yes, it's part of the original code